OSX_INSTALL_SRC.md

```diff
@@ -0,0 +1,189 @@
+    ├── isl-0.16.1/
+    ├── mpc-1.0.3/
+    └── mpfr-3.1.6/
+```
+
+### Create /usr/local/gcc-7.2
+
+> This may not be necessary.
+
+```
+sudo mkdir /usr/local/gcc-7.2
+```
+
+### Compile GMP
+
+```
+cd gmp* && mkdir build && cd build
+../configure --prefix=/usr/local/gcc-7.2 --enable-cxx
+make -j 4 && sudo make install
+rm -rf ./build
+```
+
+### Compile MPFR
+
+```
+cd ../../ && cd mpfr* && mkdir build && cd build
+../configure --prefix=/usr/local/gcc-7.2 --with-gmp=/usr/local/gcc-7.2
+make -j 4 && sudo make install
+rm -rf ./build
+```
+
+### Compile MPC
+
+```
+cd ../../ && cd mpc* && mkdir build && cd build
+../configure --prefix=/usr/local/gcc-7.2 --with-gmp=/usr/local/gcc-7.2 --with-mpfr=/usr/local/gcc-7.2
+make -j 4 && sudo make install
+rm -rf ./build
+```
+
+### Compile Graphite Optimizations Lib
+
+```
+cd ../../ && cd isl* && mkdir build && cd build
+../configure --prefix=/usr/local/gcc-7.2 --with-gmp-prefix=/usr/local/gcc-7.2
+make -j 4 && sudo make install
+rm -rf ./build
+```
+
+### Compile GCC
+
+> This will take a quite a while and turn your computer into a space heater.
+
+```
+cd ../../ && cd gcc* && mkdir build && cd build
+../configure --prefix=/usr/local/gcc-7.2 --enable-checking=release --with-gmp=/usr/local/gcc-7.2 --with-mpfr=/usr/local/gcc-7.2 --with-mpc=/usr/local/gcc-7.2 --enable-languages=c,c++ --with-isl=/usr/local/gcc-7.2 --program-suffix=-7.2
+make -j 4 && sudo make install
+rm -rf ./build
+```
+
+### Place GCC7.2 In Your Path
+
+```
+export PATH=/usr/local/gcc-7.2/bin:$PATH
+ln -s /usr/local/gcc-7.2/bin/g++-7.2 /usr/local/bin/g++-7.2
+ln -s /usr/local/gcc-7.2/bin/gcc-7.2 /usr/local/bin/gcc-7.2
+export CXX=g++-7.2
+export CC=gcc-7.2
+```
+
+### Install CMake
+
+Open ```cmake-3.9.3-Darwin-x86_64.dmg``` from Finder and follow installation instructions.
+
+From terminal:
+
+```
+sudo "/Applications/CMake.app/Contents/bin/cmake-gui" --install
+```
+
+### Install JackOSX
+
+Decompress ```JackOSX.0.92_b3.zip``` from Finder, open the resulting package, and follow installation instructions.
+
+### Install Clang 5.0
+
+Decompress ```clang+llvm-5.0.0-x86_64-apple-darwin.tar.xz```
+
+```
+sudo mkdir /usr/local/clang
+sudo mv clang+llvm-5.0.0-x86_64-apple-darwin/* /usr/local/clang
+export CXX=/usr/local/clang/bin/clang++
+export CC=/usr/local/clang/bin/clang
+export CXXFLAGS="<i have no idea what to do here to make it work>
+```
+
+
+### Clone TOP-1
+
+```
+git clone https://github.com/topisani/TOP-1.git
+```
+
+### Compile TOP-1
+
+```
+cd TOP-1
+cmake .
+```
+
+
+### References
+1. Compile gcc 7 on MacOS
+    * https://solarianprogrammer.com/2017/05/21/compiling-gcc-macos/
+1. Clang command line options
+    * https://clang.llvm.org/docs/ClangCommandLineReference.html#diagnostic-flags
+1. Getting Clang to find non-default libstdc++
+    * http://clang-developers.42468.n3.nabble.com/getting-clang-to-find-non-default-libstdc-td3945163.html
+1. Difference between -stdlib options
+    * https://stackoverflow.com/a/24419967
```

Comment:
I dont really think these instructions should be included at all - its about building gcc from scratch, not the TOP-1 or its dependencies.

Reply:
agreed.